src/backend/api/meals.js

```diff
@@ -2,11 +2,111 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+//Returns all meals
 router.get("/", async (request, response) => {
+  let maxPrice = 100000;
+  let title = "%";
+  let createdAfter = 2000;
+  let limit = (await knex("meal")).length;
+  let availableReservations = false;
+  const keyArray = Object.keys(request.query);
+
+  //for each key the respected operation related value gets
+  //assigned replacing the default values
+  keyArray.forEach((key) => {
+    switch (key) {
+      case "maxPrice":
+        maxPrice = request.query.maxPrice;
+        break;
+      case "availableReservations":
+        availableReservations = request.query.availableReservations;
+        break;
+      case "title":
+        title = request.query.title;
+        break;
+      case "createdAfter":
+        createdAfter = request.query.createdAfter;
+        break;
+      case "limit":
+        limit = request.query.limit;
+        break;
+    }
+  });
+  let filtermeal = knex("meal");
+  filtermeal
+    .where("price", "<", maxPrice)
+    .where("title", "like", "%" + title + "%")
+    .where("created_date", ">", new Date(createdAfter))
+    .limit(limit)
+    .offset(0);
+
+  //to filter as per available reservation
+  if (availableReservations) {
+    filtermeal = knex(
+      filtermeal
+        .leftJoin(
+          knex("reservation")
+            .select("meal_id")
+            .count("meal_id", { as: "reservationCount" })
+            .groupBy("meal_id")
+            .as("t1"),
+          "t1.meal_id",
+          "=",
+          "meal.id"
+        )
+        .select(
+          "meal.*",
+          knex.raw("IFNULL(reservationCount,0) as reservationCount")
+        )
+        .as("meal_reservervation")
```

Comment:
I'd save this part into it's own variable too, and name it `mealReservations`. 
Makes it easier to read and process one part at a time, it makes it also more clear that it's later being used as a sub-query.